src/tests.rs

```diff
@@ -3568,4 +3653,384 @@ mod test_module {
+                ],
+            );
+            execute_create_stream(
+                deps.as_mut(),
+                env,
+                info,
+                treasury.to_string(),
+                "test".to_string(),
+                Some("https://sample.url".to_string()),
+                in_denom.to_string(),
+                out_denom.to_string(),
+                out_supply,
+                start,
+                end,
+                Some(Decimal::from_str("0.5").unwrap()),
+            )
+            .unwrap();
+
+            // subscription
+            let mut env = mock_env();
+            env.block.height = start;
+            let funds = Coin::new(252, "in_denom");
+            let info = mock_info("subscriber", &[funds]);
+            let msg = crate::msg::ExecuteMsg::Subscribe {
+                stream_id: 1,
+                operator_target: None,
+                operator: Some("operator".to_string()),
+            };
+            let _res = execute(deps.as_mut(), env, info, msg).unwrap();
+
+            // Threshold should be reached
+            let mut env = mock_env();
+            env.block.height = end + 1;
+
+            // Exit should be possible
+            // Since there is only one subscriber all out denom should be sent to subscriber
+            // In calculations we are always rounding down that one token will be left in the stream
+            // Asuming token is 6 decimals
+            // This amount could be considered as insignificant
+            let info = mock_info("subscriber", &[]);
+            let res = execute_exit_stream(deps.as_mut(), env.clone(), info, 1, None).unwrap();
+            assert_eq!(
+                res.messages,
+                vec![SubMsg::new(BankMsg::Send {
+                    to_address: "subscriber".to_string(),
+                    amount: vec![Coin::new(499, "out_denom")],
+                })],
+            );
+
+            // Creator finalizes the stream
+            let info = mock_info("treasury", &[]);
+            let res = execute_finalize_stream(deps.as_mut(), env.clone(), info, 1, None).unwrap();
+            // Creator's revenue
+            assert_eq!(
+                res.messages[0].msg,
+                cosmwasm_std::CosmosMsg::Bank(BankMsg::Send {
+                    to_address: "treasury".to_string(),
+                    amount: vec![Coin::new(250, "in_denom")],
+                })
+            );
+            assert_eq!(
+                res.messages[1].msg,
+                cosmwasm_std::CosmosMsg::Bank(BankMsg::Send {
+                    to_address: "collector".to_string(),
+                    amount: vec![Coin::new(100, "fee")],
+                })
+            );
+            assert_eq!(
+                res.messages[2].msg,
+                cosmwasm_std::CosmosMsg::Bank(BankMsg::Send {
+                    to_address: "collector".to_string(),
+                    amount: vec![Coin::new(2, "in_denom")],
```

Comment:
We expect 2.5 as collected fee. Due to the scale of the testing values we lose 0.5 here. Best to run the test case on bigger values to not lose testing precision